File: Models/Holiday.cs
namespace MobileWebApi.Models
{
    /// <summary>
    /// Represents the Holiday table in Altroz HRMS database
    /// </summary>
    public class Holiday
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public DateTime Date { get; set; }
        public string? Description { get; set; }
        public DateTime? InsertDate { get; set; }
        public int? InsertUserId { get; set; }
        public DateTime? UpdateDate { get; set; }
        public int? UpdateUserId { get; set; }
        public int TenantId { get; set; }
        public bool IsActive { get; set; } = true;
    }
}

File: lib/feature/Home/model/holiday.dart
class Holiday {
  final int id;
  final String name;
  final DateTime date;
  final String? description;
  final int? tenantId;
  final bool isActive;

  Holiday({
    required this.id,
    required this.name,
    required this.date,
    this.description,
    this.tenantId,
    required this.isActive,
  });

  factory Holiday.fromJson(Map<String, dynamic> json) {
    return Holiday(
      id: json['id'],
      name: json['name'],
      date: DateTime.parse(json['date']),
      description: json['description'],
      tenantId: json['tenantId'],
      isActive: json['isActive'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'date': date.toIso8601String(),
      'description': description,
      'tenantId': tenantId,
      'isActive': isActive,
    };
  }
}
